src/sentry/plugins/sentry_mail/activity/release.py

```diff
@@ -1,67 +1,167 @@
 
         # identify members which have been seen in the commit log and have
         # verified the matching email address
         return {
             user: GroupSubscriptionReason.committed
             for user in User.objects.filter(
-                in_iexact('emails__email', email_list),
+                in_iexact('emails__email', self.email_list),
                 emails__is_verified=True,
-                sentry_orgmember_set__teams=project.team,
+                sentry_orgmember_set__teams=Team.objects.filter(
+                    id__in=[p.team_id for p in self.projects]
+                ),
                 is_active=True,
             ).distinct()
             if features.has('workflow:release-emails', project=self.project, actor=user)
         }
 
+    def get_users_by_teams(self):
+        if not self.user_id_team_lookup:
+            user_teams = defaultdict(list)
+            queryset = User.objects.filter(
+                sentry_orgmember_set__organization_id=self.organization.id
+            ).values('id', 'sentry_orgmember_set__teams')
+            for user_team in queryset:
+                user_teams[user_team['id']].append(user_team['sentry_orgmember_set__teams'])
+            self.user_id_team_lookup = user_teams
+        return self.user_id_team_lookup
+
     def get_context(self):
+        file_count = CommitFileChange.objects.filter(
+            commit__in=self.commit_list,
+            organization_id=self.organization.id,
+        ).values('filename').distinct().count()
+
         return {
-            'commit_list': self.commit_list,
+            'commit_count': len(self.commit_list),
+            'author_count': len(self.email_list),
+            'file_count': file_count,
+            'repos': self.repos,
             'release': self.release,
+            'deploy': self.deploy,
+            'environment': self.environment,
+            'setup_repo_link': absolute_uri('/organizations/{}/repos/'.format(
+                self.organization.slug,
+            )),
+        }
+
+    def get_user_context(self, user):
+        if user.is_superuser or self.organization.flags.allow_joinleave:
+            projects = self.projects
+        else:
+            teams = self.get_users_by_teams()[user.id]
+            projects = [p for p in self.projects if p.team_id in teams]
+        release_links = [
+            absolute_uri('/{}/{}/releases/{}/'.format(
+                self.organization.slug,
+                p.slug,
+                self.release.version,
+            )) for p in projects
+        ]
+        resolved_issue_counts = [
+            Group.objects.filter(
+                project=p,
+                id__in=GroupCommitResolution.objects.filter(
+                    commit_id__in=ReleaseCommit.objects.filter(
+                        release=self.release,
+                    ).values_list('commit_id', flat=True),
+                ).values_list('group_id', flat=True),
+            ).count() for p in projects
```

Comment:
This kinda scares me since we're doing an O(n) of `COUNT` aggregate queries. I'm pretty sure there's a way to do this in one single query, but I'm not smart enough to tell you how.